### ISSUE TITLE
"Looping" behavior while building schematic

### ISSUE THREAD

**Issue by mvrozanti** [cabaletta/baritone#2173]
## Some information
Operating system: `Linux 5.8.13-arch1-1`
Java version:
```
openjdk version "1.8.0_265"
OpenJDK Runtime Environment (build 1.8.0_265-b01)
OpenJDK 64-Bit Server VM (build 25.265-b01, mixed mode)
```

Minecraft version: 1.15.2
Baritone version: v1.5.3
Forge mods (if used): Forge is not installed; using Impact client

## Exception, error or logs
http://ix.io/2DOd

## How to reproduce
Can't really tell more than [using this schematic file](https://www.planetminecraft.com/project/spiral-sphere/)

## Modified settings
```
allowInventory
assumeStep
buildInLayers
simplifyUnloadedYCoord
```

## Final checklist
- [x] I know how to properly use check boxes 
- [x] I have included the version of Minecraft I'm running, baritone's version and forge mods (if used).
- [x] I have included logs, exceptions and / or steps to reproduce the issue.
- [] I have not used any OwO's or UwU's in this issue.

**Comment by scorbett123** [comment]
Your log file doesn't seem to include the error, is it possible that you could upload another with the error in it.

**Comment by mvrozanti** [comment]
Oh my bad here's the [latest.log](http://ix.io/2DOd). Hope this isn't a duplicate or a dumb issue :smile:

**Comment by mvrozanti** [comment]
I should add: the build started fine. A bit "loopy", in that it would glitch a bit when it found water but that was fixable by just moving around and resuming. 

But every time I start it back up (not resume) and it goes for build layer 6, the whole game crashes. When I try again, it also crashes.

**Comment by mvrozanti** [comment]
setting `simplifyUnloadedYCoord` to false seems to fix it.

**Comment by mvrozanti** [comment]
That problem seems to have fixed itself but I still get a lot of "looping" while building. It will put down a necessary block for the schematic and then break it to go somewhere and when that location is reached it builds back a path to where it put the necessary block, breaks it, and starts again.

I tried toggling buildInLayers as I read in another issue, to no avail.

**Comment by ZacSharp** [comment]
It is already penalized by having ten times the cost by default
```
/**
 * Multiply the cost of breaking a block that's correct in the builder's schematic by this coefficient
 */
public final Setting<Double> breakCorrectBlockPenaltyMultiplier = new Setting<>(10d);
```

**Comment by mvrozanti** [comment]
Oh ok I didn't know about that setting. I'm still kinda confused why it loops so much though. It happens quite frequently.

**Comment by ZacSharp** [comment]
are the blocks it breaks and places repeatedly orientation specific (e.g. chests or doors)?

**Comment by mvrozanti** [comment]
No, it's dirt, which belongs to the `acceptableThrowawayItems` setting

**Comment by ZacSharp** [comment]
and it places it in a spot where the schematic wants dirt?

**Comment by mvrozanti** [comment]
No, dirt is not a part of this schematic

**Comment by ZacSharp** [comment]
It places the dirt to get somewhere, breaks it as a wrong block and needs to go there again?

**Comment by mvrozanti** [comment]
> It places the dirt to get somewhere

Yes

> breaks it as a wrong block and needs to go there again?

I'm not able to tell

**Comment by ZacSharp** [comment]
the dirt doesn't belong there so it is a wrong block. What I meant is: it breaks the dirt before it is where it wanted to go to and then starts over

**Comment by mvrozanti** [comment]
Yes that's the behavior I'm seeing

**Comment by ZacSharp** [comment]
don't think there is a good way to prevent that right now.
Maybe try `allowBreak`, `buildIgnoreExisting` or bridging with materials it can't break with the available tools
But I doubt that any of those will work well

**Comment by mvrozanti** [comment]
Thanks I'm going to try that.

[Here's some footage of that behavior](https://streamable.com/a54s9j)

**Comment by ZacSharp** [comment]
Can you try enabling `breakFromAbove`?

**Comment by mvrozanti** [comment]
Didn't work :/

[footage](https://streamable.com/mmzdbh)

**Comment by ZacSharp** [comment]
at least it didn't waste glass.
`buildIgnoreBlocks` is my last hope. It should make baritone completely ignore the dirt

**Comment by mvrozanti** [comment]
Let me try that one

**Comment by mvrozanti** [comment]
That actually seems to work great! It leaves behind the dirt blocks but that's nothing that `mine` can't solve. Thanks a lot @ZacSharp 

I think this issue can be closed now

**Comment by ZacSharp** [comment]
that solution is just a workaround, it should be possible to build schematics in one step

**Comment by mvrozanti** [comment]
Ok so I'm gonna change the title so it's not misleading

**Comment by thecakeisalie25** [comment]
I've had this behavior too, happened while buildinlayers was set to true or false, only unique thing i can think of was that the schematic material was among the throwaway material. It also had a certain spot that it just pathfinded to for no reason every now and then, a block in the middle of the schematic it would get to no matter the cost, jump at, and then leave, sometimes getting stuck in the loop of doing that 30-40 times before just going back to normal.

**Comment by zausa** [comment]
if I set buildIgnoreBlocks to stone , it will not clear the stone like a throwaway blocks?  Or  mapart mode can clear the block that effect the mapart?

**Comment by ZacSharp** [comment]
> `buildIgnoreBlocks` is my last hope. It should make baritone completely ignore the dirt

@zausa Yes it will just ignore everything on `buildIgnoreBlocks` and you will have to clear them later (might even work to just reset `buildIgnoreBlocks` and build again in the same spot). And no, map art mode only makes it ignore blocks that are not visible on the map (i.e. covered by other blocks).

**Comment by ZacSharp** [comment]
Why did you close this? It has not been fixed yet.
